pkg/reporter/sarif_builder.go

```diff
@@ -166,3 +171,54 @@ func (b *sarifBuilder) buildFilterResultMessageMarkdown(event *analyzer.Analyzer
 
 	return msg
 }
+
+func (b *sarifBuilder) recordVulnerabilities(pkg *models.Package) {
+	if pkg.Insights == nil {
+		return
+	}
+
+	vulns := utils.SafelyGetValue(pkg.Insights.Vulnerabilities)
+	if len(vulns) == 0 {
+		return
+	}
+
+	for _, vuln := range vulns {
+		vulnId := utils.SafelyGetValue(vuln.Id)
+		if _, ok := b.violationsCache[vulnId]; ok {
+			continue
+		}
+
+		b.violationsCache[vulnId] = true
```

Comment:
`violationsCache` seems like doesn't fit here. Can we rename the `violationsCache` to only `entityCache` or something that is generic enough to hold both violation and vulns? Alternatively we can introduce `vulnCache` separately to avoid ambiguity and ID conflict resulting in subtle bugs

Reply:
yep, makes sense